mllib-local/src/test/scala/org/apache/spark/ml/util/TestingUtils.scala

```diff
@@ -32,6 +32,10 @@ object TestingUtils {
    * the relative tolerance is meaningless, so the exception will be raised to warn users.
    */
   private def RelativeErrorComparison(x: Double, y: Double, eps: Double): Boolean = {
+    // Special case for NaNs
```

Comment:
@jkbradley I do not think this change is going to be controversial, but I want to point out that from now on, matrix/vector checks will not _always_ throw errors when comparing `NaN`: the previous code would throw whenever a NaN was found.